### DISCUSSION TITLE
Ability to create layers

### DISCUSSION THREAD

**Discussion by PeteCrowe** [pubpub-zz/ppInk#228] [Ideas]
This amazing fork has so far exceeded my wildest expectations and leaves EpicPen in the digital dust. Thanks so much for the developers.

I've pretty much gone through it's extensive capabilities and was wondering how difficult it would be to add layers to it's toolset. Layers, very much like you find in any drawing package like Illustrator or Photoshop. Just like one can save drawings (and then load them), this would be more immediate and would allow layers to be stacked. It would be a great help to me.

Again, an amazing bit of coding, and thanks for the tip about getting multiple displays to work. The resolutions were the same, but for some reason, I sometimes couldn't drag the toolbar over, and when I could, I could click one button, and it would freeze up on me. The high dpi settings trick worked.

**Comment by pubpub-zz** [comment]
Hi @PeteCrowe, 
Happy you are enjoying ppInk. Do not hesitate to star the project 😊.
About your request about layers, I think that the Save/Load capability is what you are looking for : Did you tried it:
![image](https://user-images.githubusercontent.com/4083478/204606932-f581b00a-af7b-479a-9ffe-3a0020149233.png)
> _Save / Load_
through those two buttons, you will be able to store(in a text format) the current strokes. Load redraw the saved strokes onto the existing drawing.
    Save button : a long click(or first short click) is a sort of "save as" : it open the dialog box and will allow you to select the filename. the following short clicks will overwrite the strokes into the previously named file. note that a backup is done when inking is ended/quit.
Load button : a long click(or first short click) is a sort of "load as" : it open the dialog box and will allow you to select the filename. the following short clicks will load the strokes from the previously named file. at first click the file loaded is the autosave(from latest session).
Keep in mind that an automatic save is performed when closing inking mode in autosave.strokes.txt . If you have ended your drawing session by error, you can recall your work depressing load button immediately after opening session

**Reply by PeteCrowe** [reply]
Yes, I did try it, and I make mention of it in my original post! ... "Just like one can save drawings (and then load them), this would be more immediate and would allow layers to be stacked."

Sorry if I wasn't clearer. I was more thinking of the ability to pre-load these drawings onto layers (and have several layers available) and then turn them on or off with an eye icon, so that one could stack them. So, exactly what you have now, but * 3 or more. That way, when I'm recording an OBS movie, I can have them pre-loaded, and when I get to a certain topic, I can turn them on, or off as needed - even stacking them if needed.

Again, I thank you profusely to this incredibly packed, well thought out little app. I am having some minor issues, but will post that in the bug section.

**Comment by pubpub-zz** [comment]
This is what I would propose:
![image](https://user-images.githubusercontent.com/4083478/204644620-2d62b30c-80f9-4fe3-95ab-d7b7c4c53fdf.png)

the bin and eye will be reduced to small icons with other the two layer left/right icon
- left/right icon :

      short click : will move to previous/next layer. a new layer will be added automatically if reaching the end
      long click : will hide the current layer before moving to previous/next layer

- eye icon (toggle):

     short click : hide/show the current layer
     long click : hide/show all but current layers (initial stage based on very first layer)

- bin icon :

     short click : delete all strokes in current layer
     long click : delete all strokes/layers

**Comment by PeteCrowe** [comment]
Now that I see it in practice, it becomes clear that it's a bit more complicated than I had thought. Visual communication via graphics is an art form. 

I will say that I use the trash can and eye icons a whole lot, and wonder if anyone else does the same? The ones I don't use much are the recycle and snapshot. Snap shot, I have another handy app I use (on the Mac) and I just use PrtScrn (with Alt if needed) on Windows. Same with the undo. Ctrl (Cmd) + Z has been tattooed on my brain, so I never need the large undo circle arrow.

I've taken a stab at the icons (sorry for the size). Below the toolbar graphic is a simpler idea for a layers icon, but it probably isn't programmatically feasible.
![ppink icons](https://user-images.githubusercontent.com/32828222/205342966-d786850f-f813-4993-9a5a-8d57476c4726.gif)

![ppink icons](https://user-images.githubusercontent.com/32828222/205345873-c61294de-6661-433b-a702-bcbe240e422d.png)
